### PR TITLE
Batchoperation jobs only in partition 1

### LINKED ISSUE #24383: BatchOperations is only created in parition 1
BatchOperations are not run per partition and should only be done in partition 1. This means that in the `Archiver` we only should `createBatchOperationJob` if we are running on partition 1.  See [here](https://docs.google.com/document/d/184JfguLkLOp03zArBq5IX236CO9yFZY4hCh3L8Hv0-U/edit?tab=t.0).

Additionally, when calling `defaultExecutor` for partition 1 we should allocate 2 threads to the executor preform both both the `ProcessInstanceJob` and the `BatchOperationJob`. We already pass the partition id to `Archiver.create()` so the changes should not be to big.

### PR DESCRIPTION
**Pull request by rodrigo-lourenco-lopes**
## Description

This PR makes it so that we only run the `BatchOperations` in partition 1. Additionally, we allocate 2 background threads if in partition 1 so that we can have one available for both `ProcessInstanceJob` and `BatchOperationJob`.

## Checklist

- [ ] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [ ] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

## Related issues

closes https://github.com/camunda/camunda/issues/24383

### CODE REVIEW

File: zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/archiver/Archiver.java
Comment (npepinpe) [inline zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/archiver/Archiver.java:35]: :x: We should use `Protocol.START_PARTITION_ID` to be safe. I don't think we will change in the future, but better safe than sorry, and it's a very small change.

**Review by npepinpe — COMMENTED** [review]
:wrench: I didn't see any test for this. I'm not sure if it's easy to do so (e.g. a unit test in `ArchiverTest`), but ideally we should try. It's possible we would need to change the code a bit to make it easier to test :thinking: wdyt?

File: zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/archiver/Archiver.java
Comment (npepinpe) [inline zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/archiver/Archiver.java:155]: :wrench: We now have the check for `partitionId == START_PARTITION_ID` duplicated. What if we ever change the logic? How about:

- We create the jobs in a list/array. 
- The batch operation job is added only if the `partitionId == START_PARTITION_ID`.
- We pass the number of jobs to the default executor method, and we create `Math.max(jobs.size(), BG_THREAD_LIMIT)` (where `BG_THREAD_LIMIT` is some sane value - we can add a TODO about making it configurable in the future).

That way the logic is more or less in one place, and when we add more jobs, we will automatically "parallelize" them.

**Review by npepinpe — APPROVED** [review]
Regarding the testing comment, that's fair. Please address my last comment before merging :+1:

File: zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/archiver/Archiver.java
Comment (rodrigo-lourenco-lopes) [reply zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/archiver/Archiver.java:155]: Does not seem super easy to do since we have to instantiate the executor before we create the jobs which also needs to know the number of jobs 🤔 

Later, we are going to add the post archiver, which will run in all partitions, so it will not add to logic complexity from here. However, for additional jobs, I am not sure if a different logic is used. 

Do you think it is ok to merge as is now, and after the post archiver jobs decide on the limits of the number of threads and if we should use different checks here?

Comment (npepinpe) [reply zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/archiver/Archiver.java:155]: Let's do that.

**Pull request merged**
